File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/15adb3ebad729246e690688eb0dc1027.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <math.h>

#define MOD ((int)1e9 + 7)
#define BUFFER_SIZE (1 << 12)
#define MAX_LENGTH (1 << 7)

int dp[61][8];
long long n = 0;

void read(long long *arg) {
    scanf("%lld", arg);
}

void writeln(int arg) {
    printf("%d\n", arg);
}

void solve() {
    read(&n);
    memset(dp, 0, sizeof(dp));
    dp[0][0] = 1;

    for (long long i = 0; i < 60; ++i) {
        bool bit = ((1LL << i) & n);
        for (int j = 0; j < (1 << 3); ++j) {
            if (dp[i][j] == 0)
                continue;

            for (int k = ((j % 2) ^ bit); k < 8; k += 2) {
                int lol = (j + k);
                dp[i + 1][lol / 2] += dp[i][j];
                if (dp[i + 1][lol / 2] >= MOD)
                    dp[i + 1][lol / 2] -= MOD;
            }
        }
    }

    writeln(dp[60][0]);
}

int main() {
    int tests;
    read((long long *)&tests);
    while (tests--)
        solve();
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Go/15adb3ebad729246e690688eb0dc1027.go
package main

import (
	"fmt"
	"math"
)

const MOD = int(1e9 + 7)

var dp [61][8]int

func solve(n int64) {
	for i := range dp {
		for j := range dp[i] {
			dp[i][j] = 0
		}
	}

	dp[0][0] = 1
	for i := 0; i < 60; i++ {
		bit := (n & (1 << i)) != 0

		for j := 0; j < (1 << 3); j++ {
			if dp[i][j] == 0 {
				continue
			}

			for k := (j%2 ^ boolToInt(bit)); k < 8; k += 2 {
				lol := (j + k)

				dp[i+1][lol/2] += dp[i][j]
				if dp[i+1][lol/2] >= MOD {
					dp[i+1][lol/2] -= MOD
				}
			}
		}
	}

	fmt.Println(dp[60][0])
}

func boolToInt(b bool) int {
	if b {
		return 1
	}
	return 0
}

func main() {
	var tests int
	fmt.Scan(&tests)

	for i := 0; i < tests; i++ {
		var n int64
		fmt.Scan(&n)
		solve(n)
	}
}
